projects/admin.py

```diff
@@ -290,6 +309,19 @@ def view_posts_link(self, obj):
 
     view_posts_link.short_description = "Posts"
 
+    def save_formset(self, request, form, formset, change):
+        formset.save(commit=False)
+
+        for form_obj in formset.forms:
+            instance = form_obj.instance
+            if form_obj.cleaned_data.get("remove_from_project"):
+                # Remove the instance from the project
+                if isinstance(instance, Post.projects.through):
+                    instance.delete()
+            else:
+                instance.save()
+        formset.save_m2m()
```

Comment:
I don't like we override `save_formset` for the entire Project admin class for this.

I'd put custom deletion handling into own `PostProjectInlineForm`:
```python
class PostProjectInlineForm(forms.ModelForm):
    remove_from_project = forms.BooleanField(
        required=False, label="Remove from Project?", initial=False
    )

    class Meta:
        model = Post.projects.through
        fields = ["remove_from_project"]

    def save(self, commit=True):
        # Handle deletion based on the `custom_delete` field
        if self.cleaned_data.get("remove_from_project"):
            if self.instance.pk:
                self.instance.delete()
            return

        return super().save(commit=commit)
```

Reply:
Great, genuinely somehow missed that I could have just put this in the save method.